src/main/java/TaskList.java

```diff
@@ -0,0 +1,290 @@
+     * @throws IOException      Throws away the exception.
+     */
+    public String deleteTask(String inputStr) throws GrapieExceptions, IOException {
+        if (inputStr.length() <= 7) {
+            throw new GrapieExceptions(ErrorMsg.invalidNumberError);
+        } else {
+            if (!inputStr.substring(6, 7).equals(" ")) {
+                throw new GrapieExceptions(ErrorMsg.noSpaceError);
+            } else {
+                String strNumberDeleted = inputStr.substring(7, inputStr.length());
+                strNumberDeleted.replaceAll("\\s+", ""); //remove all white spaces
+
+                boolean isANumber = isNumber(strNumberDeleted);
+
+                if (isANumber) {
+                    int numToDelete = Integer.parseInt(strNumberDeleted);
+
+                    if (storingList.size() >= numToDelete) {
+
+                        int newSize = storingList.size() - 1;
+                        String toPrint = " Alrighty. I've removed this task: \n"
+                                + storingList.get(numToDelete - 1)
+                                + "\n Now you have " + newSize + " tasks in the list.";
+
+                        storingList.remove(numToDelete - 1);
+                        storage.deleteLineFromHardDisk(numToDelete);
+                        return toPrint;
+                        //formattingDivider(toPrint);
+                        //delete from hard disk
+
+
+                    } else {
+                        throw new GrapieExceptions(ErrorMsg.numberDoNotExistError(numToDelete));
+                    }
+                } else {
+                    throw new GrapieExceptions(ErrorMsg.invalidNumberError);
+                }
+            }
+        }
+    }
+
+    /**
+     * Returns the list of tasks having the user input.
+     *
+     * @param command The user input.
+     * @throws GrapieExceptions Throws grapieExceptions.
+     */
+    public String findFromList(String command) throws GrapieExceptions {
+        if (command.length() <= 5) {
+            throw new GrapieExceptions(ErrorMsg.emptyKeywordError);
+        } else {
+            //remember to add check for already completed task
+            String keyword = command.substring(5, command.length());
+
+            int counter = 1;
+            // System.out.println("    #__________________________________________________________# \n");
+            //System.out.println("    Here are the matching tasks in your list:");
+            String finalStr = "Here are the matching tasks in your list:\n";
+            for (int i = 0; i < storingList.size(); i++) {
+                Task task = storingList.get(i);
+
+                if (task.description.contains(keyword)) {
+                    finalStr += counter + "." + task + "\n";
+                    counter++;
+                }
+            }
+
+            return finalStr;
+            //System.out.println("    #__________________________________________________________# \n");
+        }
```

Comment:
I feel that Ui should handle these print statements instead. TaskList is handling too many responsibilities.